cellprofiler/modules/tram.py

```diff
@@ -0,0 +1,673 @@
+        if len(self.get_selected_tram_measurements()) == 0:
+            raise cps.ValidationError(
+                    "Please select at least one TrAM measurement for tracking of %s" % self.object_name.get_value(),
+                    self.tram_measurements)
+
+        # check on available tracking columns for the selected object
+        obj_name = self.object_name.get_value()
+        mc = pipeline.get_measurement_columns()
+        num_tracking_cols = len([entry for entry in mc if entry[0] == obj_name and entry[1].startswith(trackobjects.F_PREFIX)])
+        if num_tracking_cols == 0:
+            msg = "No %s data available for %s. Please select an object with tracking data."\
+                  % (trackobjects.F_PREFIX, obj_name)
+            raise cps.ValidationError(msg, self.object_name)
+
+    def visible_settings(self):
+        return self.settings()
+
+    def run(self, workspace):
+        """Run the module (abstract method)
+
+        workspace    - The workspace contains
+            pipeline     - instance of cpp for this run
+            image_set    - the images in the image set being processed
+            object_set   - the objects (labeled masks) in this image set
+            measurements - the measurements for this run
+            frame        - the parent frame to whatever frame is created.
+                           None means don't draw.
+            display_data - the run() module should store anything to be
+                           displayed in this attribute, which will be used in
+                           display()
+
+        run() should not attempt to display any data, but should communicate it
+        to display() via the workspace.
+        """
+        pass
+
+
+
+    def display_post_group(self, workspace, figure):
+        """Display the results of work done post-group
+
+        This method is only called if self.show_window is True
+
+        workspace - the current workspace. workspace.display_data should have
+                    whatever information is needed for the display. Numpy arrays
+                    lists, tuples, dictionaries and Python builtin objects are
+                    allowed.
+
+        figure - the figure to use for the display.
+        """
+        if self.show_window:
+            figure.set_subplots((1,1))
+            figure.subplot_histogram(0, 0, workspace.display_data.tram_values, bins=40, xlabel="TrAM",
+                                     title="TrAM for %s" % self.object_name.get_value())
+
+
+
+    def post_group(self, workspace, grouping):
+        self.show_window = True
+
+        measurements = workspace.measurements
+        obj_name = self.object_name.get_value() # the object the user has selected
+        img_numbers = measurements.get_image_numbers()
+        num_images = len(img_numbers)
+
+        def flatten_list_of_lists(lol):
+            return list(it.chain.from_iterable(lol))
```

Comment:
And use a list comprehension rather than `itertools.chain`, a quick trick for single-nested lists:

```python
list = [[1, 2], [3, 4, 5]]
sum(list, [])
> [1, 2, 3, 4, 5]
```